src/apigateway/response-validator.js

```diff
@@ -0,0 +1,36 @@
+const responseBodyValidationRuleKey = 'requiredResponse';
+
+class ResponseValidator {
+    constructor(eventClient, responseClient, schema) {
+        this._schema = schema;
+        this._responseClient = responseClient;
+        this._eventClient = eventClient;
+    }
+
+    async isValid(params) {
+        await this._checkIfRequestValid(params);
+        return this._responseClient;
+    }
+
+    async _checkIfRequestValid(params) {
+        const rule = params[responseBodyValidationRuleKey];
+
+        if (!rule) {
+            return;
+        }
```

Comment:
This logic of when to run should be handled by the router... its job is to determine what to run and when and as mentioned in the previous comment, the user of the library should be able to configure response validation by provideding a `responseBody` key.

Reply:
fixed